modules/cudaoptflow/src/nvidiaOpticalFlow.cpp

```diff
@@ -845,7 +845,7 @@ NvidiaOpticalFlowImpl_2::NvidiaOpticalFlowImpl_2(
         {
             m_enableROI = false;
             m_roiData = nullptr;
-            CV_Error(Error::StsBadFunc, "ROI not supported on this GPU");
+            // CV_Error(Error::StsBadFunc, "ROI not supported on this GPU");
```

Comment:
As I understand the final result is different (on supported and on unsupported GPUs), so this unsupported code path can't be just ignored.

/cc @vchiluka5

Reply:
https://docs.nvidia.com/video-technologies/optical-flow-sdk/nvofa-application-note/
I see. But then the application dies at this point. 
I would imagine that this check could be made as a function outside, so user could have advance notice and decide to solve it in some other way, or accept this degradation.